### PR TITLE
MLDB-1312 actually use quotechar parameter in CSV dataset and remove …

### PR DESCRIPTION
**Pull request by steadtler**
…text.line dataset

### CODE REVIEW

File: plugins/csv_dataset.cc [hunk 1]
```diff
@@ -595,63 +597,84 @@ struct CsvDataset::Itl: public TabularDataStore {
         if (config.quoter.length() == 1) {
             quote = config.quoter[0];
         }
-        else {
+        else if (config.quoter.length() > 1) {
             throw HttpReturnException(400, "Quoter string must have one character");
         }
+        else if (config.delimiter.length() != 0)
```
Comment (nicolaskruchten) [inline plugins/csv_dataset.cc]: Ideally it would be possible for the quotechar to be empty even with a delimiter.

Comment (steadtler) [inline plugins/csv_dataset.cc]: Madness! Alright I'll see what I can do.

File: plugins/text_line_dataset.cc [hunk 2]
```diff
@@ -1,297 +0,0 @@
-// This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
-
-/** text_line_dataset.cc
```
Comment (nicolaskruchten) [inline plugins/text_line_dataset.cc]: heads-up @mailletf this dataset is going away, so you will have to update the Sift notebooks if you rely on it. The replacement is text.csv.tabular with delimiter=quotechar=''.

Note #2: the text.csv.tabular dataset is queued up for being split up into a procedure and a dataset, so there's another change a'comin'.

File: plugins/csv_dataset.cc [hunk 3]
```diff
@@ -580,78 +587,100 @@ struct CsvDataset::Itl: public TabularDataStore {
...
         if (config.quoter.length() == 1) {
             quote = config.quoter[0];
+            hasQuoteChar = true;
         }
-        else {
+        else if (config.quoter.length() > 1) {
             throw HttpReturnException(400, "Quoter string must have one character");
         }
 
+        const bool isTextLine = config.quoter.length() == 0 && config.delimiter.length() == 0;
```
Comment (FinchPowers) [inline plugins/csv_dataset.cc]: `config.quoter.length() == 0` pourait pas être `config.quoter.empty()` ?
Même chose pour `config.delimiter`.

File: plugins/csv_dataset.cc [hunk 4]
```diff
@@ -580,78 +587,100 @@ struct CsvDataset::Itl: public TabularDataStore {
...
+                                           header.c_str(), header.length(), 1, 0);
+                
+                vector<string> fields = ML::expect_csv_row(pcontext, -1, separator);
 
-            switch (encoding) {
-            case ASCII:
-                for (auto & f: fields)
+                switch (encoding) {
+                case ASCII:
+                    for (auto & f: fields)
```
Comment (FinchPowers) [inline plugins/csv_dataset.cc]: Constness - Je ne suis pas certain, mais est-ce que ça pourrait être `const auto & f: fields` ?

**Comment by FinchPowers** [comment]
Commentaires mineurs à vérifier.
+1

**Pull request merged**